src/CMakeLists.txt

```diff
@@ -325,8 +325,15 @@ if(NOT TARGET RapidJSON::RapidJSON)
     INTERFACE_INCLUDE_DIRECTORIES "${CMAKE_CURRENT_SOURCE_DIR}/s3select/rapidjson/include")
 endif()
 
-option(WITH_FMT_HEADER_ONLY "use header-only version of fmt library" OFF)
-find_package(fmt 8.1.1...<10.0.0)
+option(WITH_FMT_HEADER_ONLY "use header-only version of fmt library" off)
+option(WITH_SYSTEM_FMT "allow use of system fmt if available" on)
+option(REQUIRE_SYSTEM_FMT "require use of system fmt and fail if not found" off)
```

Comment:
this complicates the building system, and adds the uncertainty.

> others might like `WITH_SYSTEM_FMT` to cause the build to fail if system fmt isn't
found.

if `WITH_SYSTEM_FMT` fails the build, then one should disable it explicitly. IMHO, building system should do exactly what is asked to do. the "guess" option makes debugging of building issues more troublesome.

Reply:
All right. I'll just `WITH_SYSTEM_FMT` and have it pass REQUIRED.